build.sh

```diff
@@ -1,20 +1,161 @@
+
+  else
+
+    # NOTE: Output log lines for stdout -> cachix caching
+    grep '^/nix/store/' "$ghcjsTmpLogFile"
+
+    # NOTE: Propagate the error state, fail the CI build
+    exit 1
+
+  fi
+
+}
+
+BUILD_PROJECT(){
 set -xe
 set -euo pipefail
 IFS=$'\n\t'
 
-GHCVERSION=${GHCVERSION:-ghc822}
-STRICT=${STRICT:-false}
-TRACING=${TRACING:-false}
-NIX_PATH=nixpkgs=https://github.com/NixOS/nixpkgs/archive/6820e2f0dd16104961d6fc7e8e38846807159c4e.tar.gz
-
-if [ "$GHCVERSION" = "ghcjs" ]; then
-    nix-build --substituters 'https://nixcache.reflex-frp.org?trusted=1' ghcjs
-else
-    nix-build                                   \
-        --argstr compiler $GHCVERSION           \
-        --arg doTracing $TRACING                \
-        --arg doStrict $STRICT                  \
-        $@
+# NOTE: If var not imported - set to the default value
+GHCVERSION=${GHCVERSION:-ghc865}
+rev=${rev:-nixpkgs-unstable}
+NIX_PATH=${NIX_PATH:-"nixpkgs=https://github.com/nixos/nixpkgs/archive/$rev.tar.gz"}
+export NIX_PATH
+name=${name:-defaultBinaryName}
+pkgName=${pkgName:-defaultPkgName}
+failOnAllWarnings=${failOnAllWarnings:-'false'}
+checkUnusedPackages=${checkUnusedPackages:-'false'}
+doCoverage=${doCoverage:-'false'}
+doHaddock=${doHaddock:-'false'}
+doJailbreak=${doJailbreak:-'false'}
+doCheck=${doCheck:-'true'}
+doBenchmark=${doBenchmark:-'false'}
+enableExecutableProfiling=${enableExecutableProfiling:-'false'}
+enableLibraryProfiling=${enableLibraryProfiling:-'false'}
+buildFromSdist=${buildFromSdist:-'false'}
+buildStrictly=${buildStrictly:-'false'}
+disableOptimization=${disableOptimization:-'true'}
+buildStackProject=${buildStackProject:-'false'}
+# NOTE: *Oprparse* key is redifined in the code further
+generateOptparseApplicativeCompletion=${generateOptparseApplicativeCompletion:-'false'}
+allowInconsistentDependencies=${allowInconsistentDependencies:-'false'}
+ghcjsTmpLogFile=${ghcjsTmpLogFile:-'/tmp/ghcjsTmpLogFile.jog'}
+ghcjsLogTailLength=${ghcjsLogTailLength:-'10000'}
+
+
+if [ "$generateOptparseApplicativeCompletion" = 'true' ]
+  then
+    # NOTE: Enable shell complition generation
+    generateOptparseApplicativeCompletion="--arg generateOptparseApplicativeCompletion $name $pkgName"
+  else
+    # NOTE: Skip the shell complition generation
+    generateOptparseApplicativeCompletion=''
+fi
+
+
+if [ "$GHCVERSION" = "ghcjs" ]
+  then
+
+    # NOTE: GHC JS build
+    # Bu itselt GHC JS build every tine creates >65000 line logs that are >4MB in size, so Travis terminates due to log size quota.
+    # nixbuild --quiet (x5) does not work on GHC JS build
+    # So there was a need to make it build.
+    # Solution - is to silence the stdout
+    # But Travis then terminates on 10 min no stdout timeout
+    # so HACK: SILENT wrapper allows to surpress the huge log, while still preserves the Cachix caching ability in any case of the build
+    # On build failure outputs the last 10000 lines of log (that should be more then enough), and terminates
```

Comment:
```suggestion
    # NOTE: GHCJS build
    # By itself, GHCJS creates >65000 lines of log that are >4MB in size, so Travis terminates due to log size quota.
    # nixbuild --quiet (x5) does not work on GHC JS build
    # So there was a need to make it build.
    # The solution is to silence the stdout
    # But Travis then terminates on 10 min no stdout timeout
    # so HACK: SILENT wrapper allows to surpress the huge log, while still preserves the Cachix caching ability in any case of the build
    # On build failure outputs the last 10000 lines of log (that should be more then enough), and terminates
```

Looks like a pretty good workaround BTW! :+1:

Reply:
Yes, it preserves all functionality we currently have.